src/Microsoft.AspNet.Server.Kestrel.Https/HttpsConnectionFilter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Server.Kestrel.Filter;
+
+namespace Microsoft.AspNet.Server.Kestrel.Https
+{
+    public class HttpsConnectionFilter : IConnectionFilter
+    {
+        private readonly X509Certificate _cert;
+        private readonly IConnectionFilter _prev;
+
+        public HttpsConnectionFilter(X509Certificate cert, IConnectionFilter prev)
+        {
+            _cert = cert;
+            _prev = prev;
+        }
+
+        public async Task OnConnection(ConnectionFilterContext context)
+        {
+            await _prev.OnConnection(context);
+
+            if (string.Equals(context.Address.Scheme, "https", StringComparison.OrdinalIgnoreCase))
```

Comment:
This should support matching certs with ports. Eventually this is where SNI will go too.